sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -166,6 +207,45 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     });
   }
 
+  /**
+   * Creates a ministry notification for the saved education
+   * program offering as a part of the same transaction.
+   * @param offeringName offering name.
+   * @param programName related program name.
+   * @param programOfferingStatus related program offering status.
+   * @param institutionLocationId related institution location id.
+   * @param institutionLocationName related institution location name.
+   * @param entityManager entity manager to be part of the transaction.
+   */
+  private async saveEducationProgramOfferingNotification(
+    offeringName: string,
+    programName: string,
+    programOfferingStatus: OfferingStatus,
+    institutionLocationId: number,
+    institutionLocationName: string,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    const institutionLocation =
+      await this.institutionLocationService.getInstitutionLocation(
+        institutionLocationId,
+      );
+    const institution = institutionLocation.institution;
+    const ministryNotification: InstitutionAddsPendingOfferingNotification = {
+      institutionName: institution.legalOperatingName,
+      institutionOperatingName: institution.operatingName,
+      institutionLocationName,
+      programName,
+      offeringName,
+      institutionPrimaryEmail: institution.primaryEmail,
+    };
+    if (programOfferingStatus === OfferingStatus.CreationPending) {
```

Comment:
Please move the `if` to the top of the method and do an early return.
If my understanding is correct, if offering is not in `CreationPending` nothing else should be checked and we can avoid, for instance, the call to `getInstitutionLocation`.